### ISSUE TITLE
veracrypt /l mounts volume under two drive letters

### ISSUE THREAD

**Issue by ivan** [veracrypt/VeraCrypt#26]
I'm mounting six volumes, each with a command like

```
"C:\Program Files\VeraCrypt\veracrypt.exe" /tc /q /p PASS /l X /v \\?\Volume{xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx}\
```

and I'm seeing VeraCrypt 1.15 (64-bit) on Windows 10 mount the volume under both `X:\` and another drive letter that I didn't request (in this case, `E:\`).

All of the "Local Disk"s in the screenshot are the redundant drive letters that I didn't request:

![veracrypt-explorer](https://cloud.githubusercontent.com/assets/4458/10182460/ee375d16-6713-11e5-9e47-1035e1829ff9.png)

This did not happen with TrueCrypt 7.1, which mounted it only under the drive letter given in the command.

I guess this is not really a huge deal, but it is odd and it does eat up the space of available drive letters.  It may also result it wasted work in programs that scan all hard drives.

**Comment by ivan** [comment]
This problem persists after rebooting.

Also, in diskmgmt.msc, I do not have the E: G: H: I: J: K: drive letters assigned to anything.

**Comment by schumann2k** [comment]
I have the same issue, mounting a volume via the GUI gives me the letter I requested plus an additional drive letter. 
Win 10, VeraCrypt 1.15 64-bit

**Comment by schumann2k** [comment]
I noticed that mounting volumes in the first available slot/letter does *not* yield two drives. Mounting a volume in any other slot creates a duplicate drive mounted on the first available letter, which does not get freed after unmounting the "real" drive.

**Comment by idrassi** [comment]
I have answered this on Codeplex: https://veracrypt.codeplex.com/workitem/226#PostedByLink18
A discussion is also their where I posted two comments: https://veracrypt.codeplex.com/discussions/645527#post1446976

All explanations are there...we are facing a difficult situation and as I explained in the links above, it may be impossible to fix the vulnerabilities without introducing side effects or changing the traditional behavior of drive letters.

A quote of my last comment. Technical 

> The Windows mount manager is the one responsible for associating \DosDevice\I with VolumeH: the big problem is that Windows mount manager seems to always try to associate the mounted volume with a local Dos device even though VeraCrypt explicitely associated this volume with the global Dos Device.
> 
> To protect against the attack, the chosen solution was to force all volumes to be mounted in the global namespace, which is what is intended in TrueCrypt and VeraCrypt. Doing so, causes the duplication since Windows mount manager create a new drive in the local namespace.
>
> At this stage, it looks like the only solution would be to mount all volumes in local namespace but this will break the traditional usage where the drive associated with a mounted TC/VC volume is accessible by all users and services on the machine.
>
> Unless a technical solution is found to block Windows mount manager from creating extra drive letters, there are two choices:
>
> * Leave the vulnerability unpatched in order to preserve the traditional TrueCrypt drive letters behavior.
> * implement a more stable fix for the vulnerability that will change the behavior of drive letter by making them local to each user on the system.
>
> Comments and technical advises are welcomed.

**Comment by E3V3A** [comment]
> implement a more stable fix for the vulnerability that will change the behavior of drive letter by making them local to each user on the system.

Seem that the second option would be the most sensible, as I fail to see why a TC/VC volume should be accessible to any other user than the one who mounted/opened it. If another user wants access, he would have the password himself and thus can mount it himself as well...

**Comment by ivan** [comment]
> Seem that the second option would be the most sensible, as I fail to see why a TC/VC volume should be accessible to any other user than the one who mounted/opened it.

This behavior is really convenient when you've got programs running under another account with `runas`.  You're still protected by the ACLs on your files, if you've set them up correctly.

> If another user wants access, he would have the password himself and thus can mount it himself as well...

You can't have two separate NTFS mounts talking to the same block device or you'd just get instant corruption.  Perhaps VeraCrypt could "steal" the existing mount from the other user, though?

**Comment by idrassi** [comment]
Just an update to write that this issue was fixed in version 1.16 as explained here: https://veracrypt.codeplex.com/discussions/645527#post1447038

Also, making the mount drive letter local instead of being global has proved to be problematic in my test as it causes issues:https://veracrypt.codeplex.com/discussions/645527#post1447411

A feature that could be implemented is support mounting to an NTFS junction point or directory link.

**Comment by asfernandes** [comment]
I'm running 1.24-Hotfix1 and still have the problem. I mount the drive on Z: but it also appear as D:
